citesphere/src/main/java/edu/asu/diging/citesphere/web/user/ItemController.java

```diff
@@ -25,34 +26,43 @@
     @Autowired
     private ICitationManager citationManager;
     
+    @Autowired
+    private SearchEngine engine;
+    
     @RequestMapping(value="/auth/group/{zoteroGroupId}/items/{itemId}")
     public String getItem(Authentication authentication, Model model, @PathVariable("zoteroGroupId") String zoteroGroupId, @PathVariable("itemId") String itemId,
-            @RequestParam(required = false, value = "index") String index, @RequestParam(defaultValue = "1", required = false, value = "page") int page,@RequestParam(value="collectionId", required=false) String collectionId,
+            @RequestParam(defaultValue = "", required = false, value = "searchTerm") String searchTerm, @RequestParam(required = false, value = "index") String index, 
+            @RequestParam(defaultValue = "1", required = false, value = "page") int page, @RequestParam(value="collectionId", required=false) String collectionId,
             @RequestParam(defaultValue = "title", required = false, value = "sortBy") String sortBy) throws GroupDoesNotExistException, CannotFindCitationException, ZoteroHttpStatusException {
         ICitation citation = citationManager.getCitation((IUser)authentication.getPrincipal(), zoteroGroupId, itemId);
         model.addAttribute("zoteroGroupId", zoteroGroupId);
         
+        CitationPage citationPage = null;
+        searchTerm.trim();
+        if (searchTerm.isEmpty()) {
+            if (index != null) {
+                citationPage = citationManager.getPrevAndNextCitation((IUser) authentication.getPrincipal(),
+                        zoteroGroupId, collectionId, page, sortBy, Integer.valueOf(index));
+            }
+        } else {
+            citationPage = engine.getPrevAndNextCitation(searchTerm, zoteroGroupId, page - 1, Integer.valueOf(index), 50);
```

Comment:
if you check for null if searchterm is empty, you would have to check here as well, dont' you? Shoudn't matter though once index has a default value.